### PR TITLE
added the ability to add a photo in the item creation modal

### PR DESCRIPTION
**Pull request by cjmielke**
## What type of PR is this?

- feature

## What this PR does / why we need it:

Adds a button to the item creation model to add a photograph, so that users don't have to dig for the option in the advanced edit menu afterwards. I've been wanting this feature for a while too, and apparently others do as well.

## Which issue(s) this PR fixes:

Implements #58 

## Special notes for your reviewer:

This is my first time developing in Vue.js, so apologies

A better solution might be to also modify the backend to accept the image on item creation, but for simplicity I just post'd a second API call from the frontend using the same function as the edit menu.

## Testing

Works on my end, and Ive done my best to make sure it works in tiny phone screens. The UI could be improved.

## Summary by CodeRabbit

- **New Features**
	- Introduced photo upload functionality in the item creation modal.
	- Added a live preview feature for selected images before submission.
	- Enhanced user interface with a new button for photo uploads.

- **Bug Fixes**
	- Improved error handling and user feedback during image upload failures.

### CODE REVIEW

**Comment by don-ferris** [comment]
Sorry for the stupid question, but is this checked in to the main branch so all I have to do is install? Or is there something more I have to do (or other branch I should use)? To me, this is core functionality, so I really want to make it happen.

**Comment by adiov** [comment]
@don-ferris You have a couple of options. The easiest is to checkout this PR's branch and then use `docker build` to build the image under a tag of your choosing. You can then modify your `docker-compose.yaml` file to use the new image you created.

@cjmielke @tankerkiller125 I tested the functionality. It works as expected, and it already made using homebox much easier and more enjoyable. The placement of the button and its styling isn't great, but I'm not a designer.

**Comment by cjmielke** [comment]
@adiov I hear ya, I'm no designer either, and I try to avoid frontend as much as I can. I tried working on that button's margins for a while, but then said "eff it, it works". CSS always fights me.

@don-ferris I just merged it to my main branch, making it a bit more straightforward. Until the devs merge it, you can get my copy up by doing the following. I'll be hosting this branch on my own eventually, once I have time to migrate my existing server.

`
git clone https://github.com/cjmielke/homebox.git
cd homebox
docker compose build
docker compose up
`

**Comment by don-ferris** [comment]
> @adiov I hear ya, I'm no designer either, and I try to avoid frontend as much as I can. I tried working on that button's margins for a while, but then said "eff it, it works". CSS always fights me.
> 
> @don-ferris I just merged it to my main branch, making it a bit more straightforward. Until the devs merge it, you can get my copy up by doing the following. I'll be hosting this branch on my own eventually, once I have time to migrate my existing server.
> 
> > git clone https://github.com/cjmielke/homebox.git
> > cd homebox
> > docker compose build
> > docker compose up

Thank you for that! When I read your reply, I was like "no way I'm going to be able to pull that off!" Way above my pay grade!

So I tried your instrux and got this:
`
 => [homebox stage-5 4/7] COPY --from=builder /go/bin/api /app             0.1s
 => [homebox stage-5 5/7] RUN chmod +x /app/api                            0.5s
 => [homebox stage-5 6/7] RUN apk add --no-cache wget                      1.2s
 => [homebox stage-5 7/7] WORKDIR /app                                     0.0s
 => [homebox] exporting to image                                           0.6s
 => => exporting layers                                                    0.6s
 => => writing image sha256:52bb1c6c21eff6d47a3eda4d6d7a0f538398e98d9b055  0.0s
 => => naming to docker.io/library/homebox                                 0.0s
 => [homebox] resolving provenance for metadata file                       0.0s
[+] Running 0/0
 ⠋ Container homebox  Recreate                                             0.0s 
Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /root/docker/homebox/data
`
No idea what to do about that. Can I get a little help?

File: frontend/components/Item/CreateModal.vue [hunk 1]
```diff
@@ -27,6 +33,21 @@
           </div>
         </div>
       </div>
+
+
+      <!-- photo preview area is AFTER the create button, to avoid pushing the button below the screen on small displays -->
+      <div class="border-t border-gray-300 p-4">
+          <!--<div class="border p-2 mt-3">-->
```
Comment (tankerkiller125) [inline frontend/components/Item/CreateModal.vue]: Can we remove these commented out code pieces?

File: frontend/components/Item/CreateModal.vue [hunk 2]
```diff
@@ -6,7 +6,13 @@
       <FormTextField ref="nameInput" v-model="form.name" :trigger-focus="focused" :autofocus="true" label="Item Name" />
       <FormTextArea v-model="form.description" label="Item Description" />
       <FormMultiselect v-model="form.labels" label="Labels" :items="labels ?? []" />
+
+
       <div class="modal-action">
+        <div class="flex">
+          <label for="photo" class="btn absolute left-0">Photo 📷</label>
```
Comment (tankerkiller125) [inline frontend/components/Item/CreateModal.vue]: Please convert the text here from hardcoded to translated.

You can do so by using `$t("file.key")` as an example, in `locales/en.json` you would put this under the `item.create_modal` key.

Thanks

**Review by tankerkiller125 — CHANGES_REQUESTED** [review]
Overall it looks good, I'm not exactly a UI dev myself, worse case this can be cleaned up a later point.

I have just two little nitpicks I'd like to see sorted before I merge this in, just some cleanup and basic translation stuff.

Thanks for the PR!

**Review by tankerkiller125 — APPROVED** [review]

**Comment by adiov** [comment]
@don-ferris Could you please edit your PR to use one of the [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)? This will assign the bounty to you.

**Comment by cjmielke** [comment]
@adiov I think you meant me - I added "closes #58" in the description, not sure if that works or not

feel free to give the bounty to @hay-kot for building something awesome!

**Comment by cjmielke** [comment]
@don-ferris now that they merged it into main, you might be able to just update your existing install

Aside from this, not sure about the error you are experiencing. Maybe you need superuser permissions on your specific system, but its just a guess.

**Comment by don-ferris** [comment]
Either I'm doing something wrong or there's a problem with the latest nightly build.
Here's my docker-compose.yml...
```
services:
  homebox:
    image: ghcr.io/sysadminsmedia/homebox:nightly
    container_name: homebox
    restart: always
    environment:
    - HBOX_LOG_LEVEL=info
    - HBOX_LOG_FORMAT=text
    - HBOX_WEB_MAX_UPLOAD_SIZE=10
    volumes:
      - ./data:/data/
    ports:
      - 3101:7745
```
And here's the command/error output:
```
Creating network "homebox_default" with the default driver
Recreating c55f516535a3_homebox ... 

ERROR: for c55f516535a3_homebox  'ContainerConfig'

ERROR: for homebox  'ContainerConfig'
Traceback (most recent call last):
  File "/usr/bin/docker-compose", line 33, in <module>
    sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
  File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 81, in main
    command_func()
  File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 203, in perform_command
    handler(command, command_options)
  File "/usr/lib/python3/dist-packages/compose/metrics/decorator.py", line 18, in wrapper
    result = fn(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1186, in up
    to_attach = up(False)
  File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 1166, in up
    return self.project.up(
  File "/usr/lib/python3/dist-packages/compose/project.py", line 697, in up
    results, errors = parallel.parallel_execute(
  File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
    raise error_to_reraise
  File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
    result = func(obj)
  File "/usr/lib/python3/dist-packages/compose/project.py", line 679, in do
    return service.execute_convergence_plan(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 579, in execute_convergence_plan
    return self._execute_convergence_recreate(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 499, in _execute_convergence_recreate
    containers, errors = parallel_execute(
  File "/usr/lib/python3/dist-packages/compose/parallel.py", line 108, in parallel_execute
    raise error_to_reraise
  File "/usr/lib/python3/dist-packages/compose/parallel.py", line 206, in producer
    result = func(obj)
  File "/usr/lib/python3/dist-packages/compose/service.py", line 494, in recreate
    return self.recreate_container(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 612, in recreate_container
    new_container = self.create_container(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 330, in create_container
    container_options = self._get_container_create_options(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 921, in _get_container_create_options
    container_options, override_options = self._build_container_volume_options(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 960, in _build_container_volume_options
    binds, affinity = merge_volume_bindings(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 1548, in merge_volume_bindings
    old_volumes, old_mounts = get_container_data_volumes(
  File "/usr/lib/python3/dist-packages/compose/service.py", line 1579, in get_container_data_volumes
    container.image_config['ContainerConfig'].get('Volumes') or {}
KeyError: 'ContainerConfig'
```
```docker-compose --version
docker-compose version 1.29.2, build unknown
```
?

**Comment by cjmielke** [comment]
@don-ferris You might be using the older "docker-compose" command, but as I learned pretty recently, the now builtin "docker compose" is the new version

https://docs.docker.com/compose/migrate/

I recently had a different issue, but that was the fix. "docker SPACE compose" instead of "docker DASH compose". The builtin command is much more up to date. If that fixes your problem, you might consider uninstalling the older docker-compose package. Thats what I did to prevent myself from ever making that mistake again.

**Comment by tankerkiller125** [comment]
That error is 100% related to the older docker-compose command which is no longer supported and is EOL. We're now on docker compose version 2 something rather using the `docker compose` command. Switching should probably solve the issue.

**Pull request merged**